libraries/adaptive-expressions/src/builtinFunctions/add.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @module adaptive-expressions
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ExpressionEvaluator, EvaluateExpressionDelegate } from '../expressionEvaluator';
+import { Expression } from '../expression';
+import { ReturnType } from '../returnType';
+import { ExpressionType } from '../expressionType';
+import { FunctionUtils } from '../functionUtils';
+
+/**
+ * Return the result from adding two or more numbers (pure number case) or concatting two or more strings (other case).
+ */
```

Comment:
It would be nice if all of these functions included examples of the possible call signatures.  When you look at the documentation, it is not at all obvious the kind or type of arguments.

Reply:
We would add more details and samples in the next stage.